build.gradle.kts

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2023 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+import com.diffplug.gradle.spotless.KotlinExtension
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import io.gitlab.arturbosch.detekt.Detekt
+import org.jetbrains.dokka.gradle.DokkaTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+  alias(libs.plugins.spotless) apply false
+  alias(libs.plugins.mavenPublish) apply false
+  alias(libs.plugins.dokka) apply false
+  alias(libs.plugins.detekt)
+  alias(libs.plugins.lint) apply false
+  alias(libs.plugins.ksp) apply false
+}
+
+val ktfmtVersion = libs.versions.ktfmt.get()
+
+allprojects {
+  apply(plugin = "com.diffplug.spotless")
+  configure<SpotlessExtension> {
+    format("misc") {
+      target("*.md", ".gitignore")
+      trimTrailingWhitespace()
+      endWithNewline()
+    }
+    kotlin {
+      target("**/*.kt")
+      ktfmt(ktfmtVersion).googleStyle()
+      trimTrailingWhitespace()
+      endWithNewline()
+      licenseHeaderFile(rootProject.file("spotless/spotless.kt"))
+      targetExclude("**/spotless.kt", "**/denylistedapis/*.kt")
+    }
+    // Externally adapted sources that should preserve their license header
+    format("kotlinExternal", KotlinExtension::class.java) {
+      target("**/denylistedapis/*.kt")
+      ktfmt(ktfmtVersion).googleStyle()
+      trimTrailingWhitespace()
+      endWithNewline()
+    }
+    kotlinGradle {
+      ktfmt(ktfmtVersion).googleStyle()
+      trimTrailingWhitespace()
+      endWithNewline()
+      licenseHeaderFile(
+        rootProject.file("spotless/spotless.kt"),
+        "(import|plugins|buildscript|dependencies|pluginManagement)"
+      )
+    }
+  }
+}
+
+subprojects {
+  pluginManager.withPlugin("java") {
+    configure<JavaPluginExtension> { toolchain { languageVersion.set(JavaLanguageVersion.of(19)) } }
+
+    tasks.withType<JavaCompile>().configureEach { options.release.set(11) }
+  }
+
+  pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+    tasks.withType<KotlinCompile>().configureEach {
+      kotlinOptions {
+        jvmTarget = "11"
+        // TODO re-enable once lint uses Kotlin 1.5
```

Comment:
What kotlin version does it use? 😱

Reply:
so this is a good one for #3. In lint 31.1.x it uses kotlin 1.7.10 (this comment is old). I'm not sure what it uses in 31.0.x or 30.4.x